### PR TITLE
[Hub Generated] Review request for Microsoft.AzureStackHCI to add version preview/2022-12-15-preview

### PR DESCRIPTION
**Pull request by pathuang-sia**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/pathuang-sia-azurestackhci-Microsoft.AzureStackHCI-2022-12-15-preview...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. April 2023
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. April 2023
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by pathuang-sia** [comment]
This merge attempt is based off of the 2022-12-15-preview api-version merged into RPSaaSMaster from this PR: https://github.com/Azure/azure-rest-api-specs-pr/pull/9996

**Review by JocelynBerrendonner — APPROVED** [review]
LGTM from a feature team's perspective

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/marketplaceGalleryImages.json [hunk 1]
```diff
@@ -0,0 +1,563 @@
...
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/MarketplaceGalleryImages"
+            }
+          },
+          "201": {
```
Comment (sjanamma) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/marketplaceGalleryImages.json:213]: Patch should not return 201 created

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/marketplaceGalleryImages.json [hunk 2]
```diff
@@ -0,0 +1,563 @@
...
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/MarketplaceGalleryImages"
+            }
+          },
+          "202": {
```
Comment (sjanamma) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/marketplaceGalleryImages.json:219]: Operation returning 202 must be a Long Running Operation

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/networkInterfaces.json [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/NetworkInterfaces"
+            }
+          },
+          "201": {
```
Comment (sjanamma) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/networkInterfaces.json:213]: Patch should not return 201 created

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/storageContainers.json [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+            }
+          },
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/StorageContainers"
+            }
+          },
+          "201": {
+            "description": "Created",
```
Comment (sjanamma) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/storageContainers.json:214]: Patch should not return 201 created

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/storageContainers.json [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+            "schema": {
+              "$ref": "#/definitions/StorageContainers"
+            }
+          },
+          "202": {
```
Comment (sjanamma) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/storageContainers.json:219]: Operations returning 202 must be a Long Running operation

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/virtualHardDisks.json [hunk 6]
```diff
@@ -0,0 +1,477 @@
...
+              "$ref": "#/definitions/VirtualHardDisks"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/VirtualHardDisks"
+            }
+          },
+          "202": {
```
Comment (sjanamma) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/virtualHardDisks.json:219]: Operations returning 202 must be a Long Running operation

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/virtualHardDisks.json [hunk 7]
```diff
@@ -0,0 +1,477 @@
...
+            }
+          },
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/VirtualHardDisks"
+            }
+          },
+          "201": {
+            "description": "Created",
```
Comment (sjanamma) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/virtualHardDisks.json:214]: Patch should not return 201 created

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/galleryImages.json [hunk 8]
```diff
@@ -0,0 +1,571 @@
...
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/GalleryImages"
+            }
+          },
+          "201": {
```
Comment (sjanamma) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/galleryImages.json:213]: Patch should not return 201 created

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/galleryImages.json [hunk 9]
```diff
@@ -0,0 +1,571 @@
...
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/GalleryImages"
+            }
+          },
+          "202": {
```
Comment (sjanamma) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/galleryImages.json:219]: Operations returning 202 must be a Long Running operation

Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/galleryImages.json:219]: will fix in next release. Work item created.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/galleryImages.json (hunk 8 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/galleryImages.json:213]: will fix in next release. Work item created.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/virtualHardDisks.json (hunk 7 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/virtualHardDisks.json:214]: will fix in next release. Work item created.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/virtualHardDisks.json (hunk 6 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/virtualHardDisks.json:219]: will fix in next release. Work item created.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/storageContainers.json (hunk 5 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/storageContainers.json:219]: will fix in next release. Work item created.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/storageContainers.json (hunk 4 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/storageContainers.json:214]: will fix in next release. Work item created.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/networkInterfaces.json (hunk 3 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/networkInterfaces.json:213]: will fix in next release. Work item created.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/marketplaceGalleryImages.json (hunk 2 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/marketplaceGalleryImages.json:219]: will fix in next release. Work item created.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/marketplaceGalleryImages.json (hunk 1 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/marketplaceGalleryImages.json:213]: will fix in next release. Work item created.

**Comment by pathuang-sia** [comment]
Backporting changes requested from ARM reviewers in this PR from: https://github.com/Azure/azure-rest-api-specs-pr/pull/12162

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/networkInterfaces.json [hunk 10]
```diff
@@ -0,0 +1,512 @@
...
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/NetworkInterfaces"
+            }
+          },
+          "202": {
+            "description": "Accepted",
+            "schema": {
+              "$ref": "#/definitions/NetworkInterfaces"
```
Comment (visingla-ms) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/networkInterfaces.json:220]: >"$ref": "#/definitions/NetworkInterfaces" [](http://example.com/codeflow?start=14&length=41)

202 can't have response schema.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/storageContainers.json [hunk 11]
```diff
@@ -0,0 +1,441 @@
...
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/StorageContainers"
+            }
+          },
+          "202": {
+            "description": "Accepted",
+            "schema": {
+              "$ref": "#/definitions/StorageContainers"
```
Comment (visingla-ms) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/storageContainers.json:220]: >"$ref": "#/definitions/StorageContainers" [](http://example.com/codeflow?start=14&length=41)

202 can't have response schema.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/virtualHardDisks.json [hunk 12]
```diff
@@ -0,0 +1,475 @@
...
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/VirtualHardDisks"
+            }
+          },
+          "202": {
+            "description": "Accepted",
+            "schema": {
+              "$ref": "#/definitions/VirtualHardDisks"
```
Comment (visingla-ms) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/virtualHardDisks.json:220]: >"$ref": "#/definitions/VirtualHardDisks" [](http://example.com/codeflow?start=14&length=40)

202 can't have response schema.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/galleryImages.json [hunk 13]
```diff
@@ -0,0 +1,569 @@
...
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/GalleryImages"
+            }
+          },
+          "202": {
+            "description": "Accepted",
+            "schema": {
+              "$ref": "#/definitions/GalleryImages"
```
Comment (visingla-ms) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/galleryImages.json:220]: >"$ref": "#/definitions/GalleryImages" [](http://example.com/codeflow?start=14&length=37)

202 can't have response schema.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/marketplaceGalleryImages.json [hunk 14]
```diff
@@ -0,0 +1,561 @@
...
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/MarketplaceGalleryImages"
+            }
+          },
+          "202": {
+            "description": "Accepted",
+            "schema": {
+              "$ref": "#/definitions/MarketplaceGalleryImages"
```
Comment (visingla-ms) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/marketplaceGalleryImages.json:220]: >"#/definitions/MarketplaceGalleryImages" [](http://example.com/codeflow?start=22&length=40)

202 can't have response schema.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/storageContainers.json (hunk 11 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/storageContainers.json]: fixed

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/galleryImages.json (hunk 13 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/galleryImages.json]: fixed

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/virtualHardDisks.json (hunk 12 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/virtualHardDisks.json]: fixed

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/marketplaceGalleryImages.json (hunk 14 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/marketplaceGalleryImages.json]: fixed

**Comment by visingla-ms** [comment]
SignedOff based the feedback by the previous ARM Reviewer.

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/arcSettings.json [hunk 15]
```diff
@@ -213,6 +262,194 @@
...
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureStackHCI/clusters/{clusterName}/arcSettings/{arcSettingName}/InitializeDisableProcess": {
+      "post": {
```
Comment (weidongxu-microsoft) [inline specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/arcSettings.json:409]: Why is this path Pascal case in last segment?

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/arcSettings.json (hunk 15 above)
Comment (pathuang-sia) [reply specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2022-12-15-preview/arcSettings.json:409]: The change was brought forth by this PR: https://github.com/Azure/azure-rest-api-specs-pr/pull/11428/files
@bniranjanbhat was there a reason?

**Pull request merged**